### ISSUE TITLE
Emit `arguments-differ` if function definition overwritten to have different argument count

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#5469]
### Current problem

Overwriting a function definition with another definition having a different argument count is a possible anti-pattern.

```python
def noop(x):  # pylint: disable=invalid-name
    """Return value unchanged"""
    return x


def add(x, y):  # pylint: disable=invalid-name
    """Add two values"""
    return x + y


def main(param):
    """Should not emit too-many-function-args"""
    tmp = noop  # I'm proposing this emit arguments-differ
    if param == 0:
        tmp = add
    return tmp(1, 1.01)
```

This code sample will become `tests/functional/t/too/too_many_function_args.py` in #5409.

### Desired solution

Emit `arguments-differ` in the marked line above.

### Additional context

Discussed in https://github.com/PyCQA/pylint/pull/5409/files/13ebc62da361bd4856af9b7c1daccc69b7210e15#r762428183.
